src/codelabs/iterables.md

```diff
@@ -0,0 +1,615 @@
+    Iterable<int> iterable = [1, 2, 3];
+    var value = iterable.elementAt(1);
+{% endprettify %}
+
+Continue to the next section to learn more about how to access elements of an `Iterable`.
+
+## Reading elements
+
+The elements of an `Iterable` can be read sequentially.
+
+One way to do that, is by getting an `Iterator` using the `iterator` getter,
+and using it to step through the values. However, it is easier to use the 
+`for-in loop`, which internally uses an `Iterator` too.
+
+### Example: Using for-in loop
+
+The following example shows how to read elements using  a `for-in loop`.
+
+[//]: https://gist.github.com/597f152c1df414fc4d7d78d1be22ac22
+<iframe
+  src="{{site.dartpad-embed}}?id=597f152c1df414fc4d7d78d1be22ac22"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="200"
+  width="100%" >
+</iframe>
+
+The `for-in loop`, as opposed to the `Iterator`, makes your code easier
+to read and understand, and it is less prone to errors.
+
+Here's an example with `Iterator`:
+
+{% prettify dart %}
+var it = iterable.iterator;	
+while (it.moveNext()) {	
+  print(it.current);	
+}
+{% endprettify %}
+
+{{site.alert.info}}
+Stepping with the `Iterator` is done
+by calling `Iterator.moveNext`, and if the call returns true, the `Iterator` 
+has now moved to the next element, which is then available as `Iterator.current`. 
+
+If the call returns `false`, there are no more elements, and `iterator.current` returns `null`.
+{{site.alert.end}}
+
+{{site.alert.secondary}}
+**Key terms:**
+* **Iterable**: the Dart [Iterable][iterable class] class.
+* **Iterator**: used to read elements from an Iterable.
+* **for-in loop**: an easy way to sequentially read elements from an Iterable.
+{{site.alert.end}}
+
+### Example: Using first and last
+
+In some cases, you only want to access the first or the last element of an `Iterable`.
+
+With the `Iterable` class you can't access to the elements directly, so you can't call
+to `iterable[0]` to access the first element. Instead, you can use the method `first`
+which will return the first element.
+
+As well, you can't use the operator `[]` to access the last element, but you can use the
+method `last`. 
+
+{{site.alert.warn}}
+  Because accessing the last element of an iterable requires stepping through all the other elements,
+  **`last` can be slow.**
+{{site.alert.end}}
+
+Accessing to the `first` or `last` element of an empty `Iterable` will throw a 
+[StateError][StateError class].
+
+[//]: https://gist.github.com/f3d96039195566d934a966390ea4ad62
+<iframe
+  src="{{site.dartpad-embed}}?id=f3d96039195566d934a966390ea4ad62"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="300"
+  width="100%" >
+</iframe>
+
+In this example you have seen how `first` and `last` can be used to access the first
+and the last elements of an `Iterator`, but would it be possible to use a similar method
```

Comment:
`Iterator` -> iterable

(or `Iterable` object, or whatever convention we decide on. I'm thinking of following the lead of `Future` and using _`Iterable`_ for the class and _iterable_ for instances of that class. We'd probably want to set up this terminology in a "Key terms" note, like the async-await codelab does.)

Reply:
I mean Iterable, yes :)